src/util/server.c

```diff
@@ -38,6 +38,13 @@
 #include <sys/prctl.h>
 #endif
 
+static TALLOC_CTX *autofree_ctx;
+
+static void server_atexit(void)
+{
+    talloc_zfree(autofree_ctx);
```

Comment:
I wonder why original code in libtalloc uses complicated logic with d-tor setting `autofree_ctx` to 0 instead of zfree...

Reply:
I believe that is to not break it if someone calls talloc_free(talloc_get_autofree_context()).